### PR TITLE
Drop perma-fail ci-kubernetes-e2e-gce-taint-evict CI job

### PR DESCRIPTION
**Pull request by dims**
`TaintBasedEvictions` went GA a long time ago (1.19) so this job has
been failing since then because the feature gate no longer exists!

See test-grid : https://testgrid.k8s.io/google-gce#gce-taint-evict

From https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-taint-evict/1516438770801446912 you can see that the manager just keeps looping and eventually fails.
```
bootstrap-e2e-master/kube-controller-manager.log:Error: invalid argument "TaintBasedEvictions=true" for "--feature-gates" flag: unrecognized feature gate: TaintBasedEvictions
bootstrap-e2e-master/kube-controller-manager.log:Error: invalid argument "TaintBasedEvictions=true" for "--feature-gates" flag: unrecognized feature gate: TaintBasedEvictions
bootstrap-e2e-master/kube-controller-manager.log:Error: invalid argument "TaintBasedEvictions=true" for "--feature-gates" flag: unrecognized feature gate: TaintBasedEvictions
bootstrap-e2e-master/kube-controller-manager.log:Error: invalid argument "TaintBasedEvictions=true" for "--feature-gates" flag: unrecognized feature gate: TaintBasedEvictions
bootstrap-e2e-master/kube-controller-manager.log:Error: invalid argument "TaintBasedEvictions=true" for "--feature-gates" flag: unrecognized feature gate: TaintBasedEvictions
bootstrap-e2e-master/kube-controller-manager.log:Error: invalid argument "TaintBasedEvictions=true" for "--feature-gates" flag: unrecognized feature gate: TaintBasedEvictions
bootstrap-e2e-master/kube-controller-manager.log:Error: invalid argument "TaintBasedEvictions=true" for "--feature-gates" flag: unrecognized feature gate: TaintBasedEvictions
bootstrap-e2e-master/kube-controller-manager.log:Error: invalid argument "TaintBasedEvictions=true" for "--feature-gates" flag: unrecognized feature gate: TaintBasedEvictions
bootstrap-e2e-master/kube-controller-manager.log:Error: invalid argument "TaintBasedEvictions=true" for "--feature-gates" flag: unrecognized feature gate: TaintBasedEvictions
bootstrap-e2e-master/kube-controller-manager.log:Error: invalid argument "TaintBasedEvictions=true" for "--feature-gates" flag: unrecognized feature gate: TaintBasedEvictions
bootstrap-e2e-master/kube-controller-manager.log:Error: invalid argument "TaintBasedEvictions=true" for "--feature-gates" flag: unrecognized feature gate: TaintBasedEvictions
```

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

### CODE REVIEW

**Comment by dims** [comment]
/assign @soltysh 

(as this periodic job seems to be under sig-apps)

**Comment by BenTheElder** [comment]
> TaintBasedEvictions went GA a long time ago (1.19) so this job has
> been failing since then because the feature gate no longer exists!

👏 👏 
/approve
/lgtm

File: config/jobs/kubernetes/sig-apps/sig-apps-config.yaml [hunk 1]
```diff
@@ -1,30 +1,4 @@
...
-      - --gcp-node-image=gci
-      - --gcp-zone=us-west1-b
-      - --provider=gce
-      - --test_args=--ginkgo.focus=\[Feature:TaintEviction\] --minStartupPods=8
-      - --timeout=50m
-      image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20220413-ebb00ad7a6-master
-
-
-  annotations:
-    testgrid-dashboards: google-gce
```
Comment (BenTheElder) [inline config/jobs/kubernetes/sig-apps/sig-apps-config.yaml:26]: aside: really need to nuke most of these google-* dashboards entirely, they're really not meaningful at this point

Comment (dims) [reply config/jobs/kubernetes/sig-apps/sig-apps-config.yaml:26]: for sure!

**Comment by ameukam** [comment]
/lgtm
🚀

**Pull request merged**